### PR TITLE
test(werklijst): add E2E coverage for urgentie badges and status

### LINKED ISSUE #509: 🛠 Task: E2E tests werklijst urgentie-status (Phase 2 — Feature #402)
# 🛠 Task: E2E tests voor werklijst urgentie-status en verbeterde kolommen

---

## Intent

Schrijf en voer Playwright E2E tests uit die alle Gherkin scenarios van Feature #402 afdekken, conform de [QA.md Phase 2 workflow](docs/steering/QA.md#phase-2-e2e-verification-post-deploy-to-test-environment). Dit is de laatste stap om de feature volledig te verifiëren.

---

## Functional Description

Na deploy van Feature #402 naar `ita.test.icatt.nl` moeten geautomatiseerde E2E tests bewijzen dat urgentie-badges, extra kolommen, en klikbare rijen correct functioneren voor echte gebruikers.

---

## Design Reference

N/A — geen design voor deze task (test-only).

---

### In scope

- Playwright E2E tests voor alle Gherkin scenarios uit #504, #505 en #506
- Tests draaien tegen `ita.test.icatt.nl`
- Rapportage pass/fail per scenario

### Out of scope

- Implementatiewijzigingen aan de feature zelf
- Unit tests (niet van toepassing per QA.md)

---

## Contracts & Interfaces

N/A — deze task produceert alleen testcode.

---

## Technical Approach

- **Test file:** `InterneTaakAfhandeling.EndToEndTest/Werklijst/WerklijstUrgentieScenarios.cs` — `[TestClass][DoNotParallelize] class WerklijstUrgentieScenarios : ITAPlaywrightTest`, mirroring `ContactverzoekScenarios.cs`'s `Step()`/`RegisterCleanup` pattern.
- **TestDataHelper:** add an optional `DateTime? plaatsgevondenOp` parameter to the private `CreateContactmoment(...)` (currently hardcodes `UtcNow`) so tests can create contactverzoeken with an arbitrary past ContactDatum — needed to trigger green/orange/red/day-format states. Existing call sites unaffected (defaults to `UtcNow`).
- **New test-only `BusinessHours` helper:** mirrors the app's Mon–Fri business-hour arithmetic, used only to compute deterministic `ContactDatum` values for test setup. Assertions still target the real rendered badge/label — this doesn't fake the thing under test, only the fixture data.
- **Locators.cs:** add `GetColumnHeader(name)`, `GetUrgentieBadge(onderwerp)` (scoped by row — no ARIA role/testid exists on the `utrecht-badge-status` element, so it's located via row-scoped text/CSS class). Reuse existing `GetDetailsLink` for the arrow-link/aria-label scenarios (already migrated off "Klik hier").
- **Data setup:** contactverzoeken created with computed `ContactDatum` via the new `TestDataHelper` overload to trigger each threshold.
- **Trade-off — weekend scenario:** anchor `ContactDatum` to "the most recent Friday 17:00 before now" rather than requiring the suite to literally run on a Monday — makes weekend-exclusion assertion deterministic on any day of the week.
- **Trade-off — keyboard scenario:** asserts against the `<a>` link, not the `<tr>` — confirmed in code that only the link is focusable/keyboard-operable (matches WCAG guidance in TECH.md); the row itself has no `tabindex`/keydown handler.
- **Execution constraint:** these tests require Azure AD + TOTP login against a live `ita.test.icatt.nl` (per QA.md). This session writes and build-verifies (`dotnet build`) all 12 scenarios; actual red/green execution against the deployed test environment happens outside this session (developer machine or CI).

### Aggregates & Invariants

N/A — test-only task, geen domeinwijzigingen.

---

## Impacted Areas

- Backend: geen wijzigingen
- Frontend: geen wijzigingen
- Test: `InterneTaakAfhandeling.EndToEndTest/Werklijst/WerklijstUrgentieScenarios.cs` (nieuw), `InterneTaakAfhandeling.EndToEndTest/Infrastructure/Locators.cs` (uitgebreid), `InterneTaakAfhandeling.EndToEndTest/Infrastructure/TestDataHelper.cs` (uitgebreid met datum-parameter)
- APIs: leest bestaande endpoints
- External systems: `ita.test.icatt.nl` als test target

---

## Gherkin

Scenarios afkomstig uit implementatietasks — te verifiëren via E2E:

```gherkin
Feature: Werklijst urgentie-status en verbeterde kolommen

  # Uit Task #504 — Urgentie-statuslabel
  Scenario: Contactverzoek binnen afhandeltermijn toont groen label
    Given een contactverzoek met ContactDatum minder dan 42 werkdaguren geleden
    When de behandelaar de werklijst opent
    Then toont het urgentielabel een groene badge met "nog Xu"

  Scenario: Contactverzoek bijna verlopen toont oranje label
    Given een contactverzoek met ContactDatum zodanig dat minder dan 6 werkdaguren resteren
    When de behandelaar de werklijst opent
    Then toont het urgentielabel een oranje badge met "nog Xu"

  Scenario: Contactverzoek verlopen toont rood label
    Given een contactverzoek met ContactDatum meer dan 48 werkdaguren geleden
    When de behandelaar de werklijst opent
    Then toont het urgentielabel een rode badge met "Xu verlopen"

  Scenario: Weekenden uitgesloten van berekening
    Given een contactverzoek aangemaakt op vrijdag 17:00 met 48u termijn
    When de behandelaar de werklijst opent op maandag 09:00
    Then is de urgentie nog steeds groen (weekend niet meegeteld)

  Scenario: Ceiling-afronding voorkomt "0u" label
    Given een contactverzoek met minder dan 1 werkdaguur resterend
    When de behandelaar de werklijst opent
    Then toont het label "nog 1u" (niet "nog 0u")

  Scenario: Urgentie zichtbaar op beide werklijstweergaven
    Given een contactverzoek toegewezen aan de behandelaar
    When de behandelaar "Mijn werkvoorraad" en "Alle contactverzoeken" opent
    Then is het urgentielabel zichtbaar op beide tabbladen

  # Uit Task #505 — Kolommen Afdeling en Klantcontactnummer
  Scenario: Kolom Afdeling zichtbaar op Mijn werkvoorraad
    Given een contactverzoek met een bekende afdeling
    When de behandelaar "Mijn werkvoorraad" opent
    Then is de kolom "Afdeling" zichtbaar met de juiste waarde

  Scenario: Kolom Klantcontactnummer zichtbaar op alle werklijsten
    Given een contactverzoek met een klantcontactnummer
    When de behandelaar een werklijst opent
    Then is de kolom "Klantcontactnummer" zichtbaar met het nummer

  Scenario: Contactverzoek zonder afdeling toont lege cel
    Given een contactverzoek zonder gekoppelde afdeling
    When de behandelaar "Mijn werkvoorraad" opent
    Then toont de kolom Afdeling een lege cel of "-"

  # Uit Task #506 — Klikbare rijen
  Scenario: Klikbare rij opent contactverzoek
    Given een contactverzoek op de werklijst
    When de behandelaar op de rij klikt
    Then navigeert de browser naar de detailpagina van het contactverzoek

  Scenario: Rij toont hover-effect en pointer
    Given een contactverzoek op de werklijst
    When de behandelaar met de muis over de rij beweegt
    Then verandert de cursor naar pointer en toont de rij een achtergrondkleur

  Scenario: Link in rij toont pijl in plaats van "Klik hier"
    Given een contactverzoek op de werklijst
    When de behandelaar de Details-kolom bekijkt
    Then toont de cel een "→" link met een beschrijvend aria-label
```

---

## Edge Cases & Risks

- Testdata timing: urgentie-berekening is tijdsafhankelijk — tests moeten contactverzoeken aanmaken met berekende `ContactDatum` relatief aan het moment van uitvoering
- Weekend-scenario is lastig deterministisch te testen — overweeg een vaste datum via testdata of accepteer dat dit scenario alleen op bepaalde dagen groen/rood oplevert
- Flakiness door sessie-timeouts (bekend probleem, zie QA.md Known Flaky Areas)

---

## Observability

None required for this task — test-only.

---

## Dependencies

- Depends on: Feature #402 merged naar main en deployed naar `ita.test.icatt.nl`
- Depends on: Tasks #504, #505, #506 verified

---

## Rollout

- Feature flag: N/A
- Backward compatibility: N/A
- Data migration: N/A

---

## Definition of Done

- [x] Alle 12 Gherkin scenarios gedekt door Playwright E2E tests
- [x] Tests passen tegen `ita.test.icatt.nl` (12/12, verified with 2 full re-runs after fixing 2 flaky/racy assertions)
- [x] Geen nieuwe flaky tests geïntroduceerd
- [x] Edge cases (lege afdeling, geen klantcontactnummer) gedekt
- [ ] Code reviewed
- [ ] PR gemerged naar main

---

## Test Mapping

All 12 scenarios executed against the live `ita.test.icatt.nl` environment (12/12 passing).

| Gherkin Scenario | Test file | Status |
| --- | --- | --- |
| Groen urgentielabel | `Werklijst/WerklijstUrgentieScenarios.cs:39` | passing |
| Oranje urgentielabel | `Werklijst/WerklijstUrgentieScenarios.cs:56` | passing |
| Rood urgentielabel | `Werklijst/WerklijstUrgentieScenarios.cs:73` | passing |
| Weekenden uitgesloten | `Werklijst/WerklijstUrgentieScenarios.cs:90` | passing |
| Ceiling-afronding | `Werklijst/WerklijstUrgentieScenarios.cs:114` | passing |
| Urgentie op beide weergaven | `Werklijst/WerklijstUrgentieScenarios.cs:135` | passing |
| Kolom Afdeling | `Werklijst/WerklijstUrgentieScenarios.cs:156` | passing |
| Kolom Klantcontactnummer | `Werklijst/WerklijstUrgentieScenarios.cs:176` | passing |
| Lege afdeling | `Werklijst/WerklijstUrgentieScenarios.cs:198` | passing |
| Klikbare rij | `Werklijst/WerklijstUrgentieScenarios.cs:217` | passing |
| Hover-effect | `Werklijst/WerklijstUrgentieScenarios.cs:236` | passing |
| Pijl link | `Werklijst/WerklijstUrgentieScenarios.cs:269` | passing |

### PR DESCRIPTION
**Pull request by surbhijha92**
## Summary

Feature #402 ("Werklijst toont urgentie-status en verbeterde kolommen") relies on E2E tests as its only verification layer (per `docs/steering/QA.md` — no unit/integration tests exist in this project). Its three implementation tasks (#498, #499, #503) were merged and code-inspection-verified in Phase 1, but per QA.md's two-phase test strategy, the feature isn't fully proven until Phase 2: real Playwright tests running against a deployed environment. This PR is that Phase 2 — it writes and executes the E2E coverage that finally proves the urgentie badges, new columns, and clickable rows work for real users.

## Changes

- **`BusinessHours.cs`** (new): mirrors `UrgentieBerekenService`'s Mon–Fri, 24h/day business-hour arithmetic. Used to (a) build a `ContactDatum` a target number of business hours in the past regardless of which weekday the suite runs, and (b) compute the expected badge status/label at assertion time as a "known answer" oracle, so drift between fixture creation and page load can't produce a flaky assertion.
- **`TestDataHelper.cs`**: adds `CreateContactverzoekWithContactDatum` (accepts an explicit `PlaatsgevondenOp`) and `CreateContactverzoekWithCurrentUserAssignedNoAfdeling` (for the empty-Afdeling-cell edge case). The underlying `CreateContactmoment` now accepts an optional date instead of hardcoding `UtcNow`.
- **`Locators.cs`**: adds werklijst-table locators — `GetWerklijstRow`, `GetColumnHeader`, `GetUrgentieBadge`, and `GetColumnCellAsync` (resolves a row's cell by matching column header text at runtime, since Afdeling/Klantcontactnummer sit at different indices on the two tables).
- **`WerklijstUrgentieScenarios.cs`** (new): all 12 Gherkin scenarios from tasks #498/#499/#503, executed and passing against `ita.test.icatt.nl`.

No application code changed — this is a test-only task per its scope.

## Test plan

- [x] Contactverzoek binnen afhandeltermijn toont groen label
- [x] Contactverzoek bijna verlopen toont oranje label
- [x] Contactverzoek verlopen toont rood label
- [x] Weekenden uitgesloten van berekening (fixture deliberately spans >5 weekdays to force at least one weekend into the calculation, regardless of run day)
- [x] Ceiling-afronding voorkomt "0u" label
- [x] Urgentielabel zichtbaar op beide werklijstweergaven
- [x] Kolom Afdeling zichtbaar op Mijn werkvoorraad
- [x] Kolom Klantcontactnummer zichtbaar op alle werklijsten
- [x] Contactverzoek zonder afdeling toont lege cel
- [x] Klikbare rij opent contactverzoek
- [x] Rij toont hover-effect en pointer
- [x] Link in rij toont pijl in plaats van "Klik hier"

All 12 run and pass together against the live test environment (`dotnet test --filter FullyQualifiedName~WerklijstUrgentieScenarios`, ~3.3 min). Two issues surfaced and were fixed during this run: `GetColumnCellAsync` didn't wait for the table to render before scanning headers (fixed by waiting on the row first), and the hover test's "before" baseline could inherit mouse position from a prior test's click (fixed by explicitly moving the mouse away first).

## Related

Closes #509

### CODE REVIEW

File: InterneTaakAfhandeling.EndToEndTest/Infrastructure/TestDataHelper.cs [hunk 1]
```diff
@@ -169,6 +169,65 @@ await CreateInternetaak(
...
+                "This is a test contact request created during an end-to-end test run.",
+                klantnaam: null,
+                plaatsgevondenOp: plaatsgevondenOp);
+
+            var submitterActor = await GetOrCreateSubmitterActor();
+            await ConnectActorToContactmoment(submitterActor, contactmoment.Uuid);
+
+            var afdelingActor = await GetOrCreateAfdelingActor(afdelingKey);
+            var medewerkerActor = await GetOrCreateMedewerkerActor("icatt-integratie-test@icatt.nl");
+
```
[the remark this answers is not in the document]
Comment (surbhijha92) [reply InterneTaakAfhandeling.EndToEndTest/Infrastructure/TestDataHelper.cs:193]: Fixed

File: InterneTaakAfhandeling.EndToEndTest/Infrastructure/TestDataHelper.cs [hunk 2]
```diff
@@ -169,6 +169,65 @@ await CreateInternetaak(
...
+            var contactmoment = await CreateContactmoment(
+                onderwerp,
+                "This is a test contact request created during an end-to-end test run.",
+                klantnaam: null);
+
+            var submitterActor = await GetOrCreateSubmitterActor();
+            await ConnectActorToContactmoment(submitterActor, contactmoment.Uuid);
+
+            var medewerkerActor = await GetOrCreateMedewerkerActor("icatt-integratie-test@icatt.nl");
+
```
[the remark this answers is not in the document]
Comment (surbhijha92) [reply InterneTaakAfhandeling.EndToEndTest/Infrastructure/TestDataHelper.cs:222]: Fixed

File: InterneTaakAfhandeling.EndToEndTest/Werklijst/WerklijstUrgentieScenarios.cs [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+
+            var assertionTime = DateTime.UtcNow;
+            var elapsed = BusinessHours.ElapsedBusinessHours(contactDatum, assertionTime);
+            var resterend = BusinessHours.AfhandeltermijnUren - elapsed;
+            var expectedStatusClass = BusinessHours.ExpectedStatusClass(resterend);
+            var expectedLabel = BusinessHours.ExpectedLabel(resterend);
+
+            await Step($"Verify badge has status class '{expectedStatusClass}' and label '{expectedLabel}'");
+            await Expect(badge).ToHaveClassAsync(new Regex(Regex.Escape(expectedStatusClass)));
+            await Expect(badge).ToHaveTextAsync(expectedLabel);
```
[the remark this answers is not in the document]
Comment (surbhijha92) [reply InterneTaakAfhandeling.EndToEndTest/Werklijst/WerklijstUrgentieScenarios.cs]: Fixed

File: InterneTaakAfhandeling.EndToEndTest/Werklijst/WerklijstUrgentieScenarios.cs [hunk 4]
```diff
@@ -0,0 +1,309 @@
...
+namespace InterneTaakAfhandeling.EndToEndTest.Werklijst
+{
+    [TestClass]
+    [DoNotParallelize]
+    public class WerklijstUrgentieScenarios : ITAPlaywrightTest
+    {
+        private const string MijnWerkvoorraadPath = "/";
+        private const string AlleContactverzoekenPath = "/alle-contactverzoeken";
+
+        private async Task GotoAsync(string path)
```
Comment (felixcicatt) [inline InterneTaakAfhandeling.EndToEndTest/Werklijst/WerklijstUrgentieScenarios.cs:14]: why do we need this? can you explain in a comment? why does navigating fail (sometimes)?

File: InterneTaakAfhandeling.EndToEndTest/Werklijst/WerklijstUrgentieScenarios.cs [hunk 5]
```diff
@@ -0,0 +1,309 @@
...
+
+            await Step("Verify the link has a descriptive aria-label");
+            var ariaLabel = await detailsLink.GetAttributeAsync("aria-label");
+            Assert.IsNotNull(ariaLabel);
+            Assert.IsTrue(ariaLabel.StartsWith("Open contactverzoek", StringComparison.Ordinal), $"aria-label should describe the contactverzoek, was '{ariaLabel}'");
+        }
+
+        // --- Shared assertion helper ---
+
+        private async Task AssertUrgentieBadgeMatchesOracle(string onderwerp, DateTime contactDatum)
```
Comment (felixcicatt) [inline InterneTaakAfhandeling.EndToEndTest/Werklijst/WerklijstUrgentieScenarios.cs]: what does Oracle mean in this case?

File: InterneTaakAfhandeling.EndToEndTest/Werklijst/BusinessHours.cs [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+    /// (afhandeltermijn 48h, bijna-verlopen drempel 6h). Used to:
+    /// <list type="bullet">
+    /// <item>construct a ContactDatum a target number of business hours in the past, immune to
+    /// which weekday the suite happens to run on (<see cref="SubtractBusinessHours"/>)</item>
+    /// <item>compute the expected badge status/label at assertion time as a "known answer"
+    /// oracle, since real elapsed time drifts slightly between fixture creation and page load
+    /// (<see cref="ElapsedBusinessHours"/>, <see cref="ExpectedStatusClass"/>, <see cref="ExpectedLabel"/>)</item>
+    /// </list>
+    /// </summary>
+    internal static class BusinessHours
```
Comment (mstokericatt) [inline InterneTaakAfhandeling.EndToEndTest/Werklijst/BusinessHours.cs:8]: this seems like a copy of the code used by ita. if so, can't we share it instead of duplicating it?

Comment (felixcicatt) [reply InterneTaakAfhandeling.EndToEndTest/Werklijst/BusinessHours.cs:8]: but then, what are we testing?

Comment (mstokericatt) [reply InterneTaakAfhandeling.EndToEndTest/Werklijst/BusinessHours.cs:8]: 😅 well, we are not testing the correct calculation of the businesshours here. Doing that by duplicating the code and seeing if it has the same outcome would be insane anyway

Comment (felixcicatt) [reply InterneTaakAfhandeling.EndToEndTest/Werklijst/BusinessHours.cs:8]: agreed :-) can we make the test useful without assuming what the number of business hours should be?

File: InterneTaakAfhandeling.EndToEndTest/Werklijst/WerklijstUrgentieScenarios.cs (hunk 5 above)
Comment (surbhijha92) [reply InterneTaakAfhandeling.EndToEndTest/Werklijst/WerklijstUrgentieScenarios.cs]: oracle was a jargon testing term. I replaced it with AssertUrgentieBadgeMatchesExpected

File: InterneTaakAfhandeling.EndToEndTest/Werklijst/BusinessHours.cs (hunk 6 above)
Comment (surbhijha92) [reply InterneTaakAfhandeling.EndToEndTest/Werklijst/BusinessHours.cs:8]: Done. Removed the duplicated ElapsedBusinessHours, ExpectedStatusClass and ExpectedLabel methods. Each test now directly asserts the status bucket it was designed for (e.g. 10h elapsed → success, 53h → error), with the reasoning in the comments. SubtractBusinessHours is kept only to place the fixture at the right point in time, skipping weekends.

File: InterneTaakAfhandeling.EndToEndTest/Werklijst/WerklijstUrgentieScenarios.cs (hunk 4 above)
Comment (surbhijha92) [reply InterneTaakAfhandeling.EndToEndTest/Werklijst/WerklijstUrgentieScenarios.cs:14]: Azure AD sessions can intermittently time out mid-suite, and navigation can hit a transient ERR_ABORTED when a frame is still settling. Without this, a one-off hiccup fails the whole test

**Review by felixcicatt — APPROVED** [review]
